api/v1beta1/azuremanagedcontrolplanetemplate_webhook.go

```diff
@@ -0,0 +1,289 @@
+		allErrs = append(allErrs, err)
+	}
+
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("Spec", "Template", "Spec", "DNSServiceIP"),
+		old.Spec.Template.Spec.DNSServiceIP,
+		mcp.Spec.Template.Spec.DNSServiceIP); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("Spec", "Template", "Spec", "NetworkPlugin"),
+		old.Spec.Template.Spec.NetworkPlugin,
+		mcp.Spec.Template.Spec.NetworkPlugin); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("Spec", "Template", "Spec", "NetworkPolicy"),
+		old.Spec.Template.Spec.NetworkPolicy,
+		mcp.Spec.Template.Spec.NetworkPolicy); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("Spec", "Template", "Spec", "LoadBalancerSKU"),
+		old.Spec.Template.Spec.LoadBalancerSKU,
+		mcp.Spec.Template.Spec.LoadBalancerSKU); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if old.Spec.Template.Spec.AADProfile != nil {
+		if mcp.Spec.Template.Spec.AADProfile == nil {
+			allErrs = append(allErrs,
+				field.Invalid(
+					field.NewPath("Spec", "Template", "Spec", "AADProfile"),
+					mcp.Spec.Template.Spec.AADProfile,
+					"field cannot be nil, cannot disable AADProfile"))
+		} else {
+			if !mcp.Spec.Template.Spec.AADProfile.Managed && old.Spec.Template.Spec.AADProfile.Managed {
+				allErrs = append(allErrs,
+					field.Invalid(
+						field.NewPath("Spec", "Template", "Spec", "AADProfile.Managed"),
+						mcp.Spec.Template.Spec.AADProfile.Managed,
+						"cannot set AADProfile.Managed to false"))
+			}
+			if len(mcp.Spec.Template.Spec.AADProfile.AdminGroupObjectIDs) == 0 {
+				allErrs = append(allErrs,
+					field.Invalid(
+						field.NewPath("Spec", "Template", "Spec", "AADProfile.AdminGroupObjectIDs"),
+						mcp.Spec.Template.Spec.AADProfile.AdminGroupObjectIDs,
+						"length of AADProfile.AdminGroupObjectIDs cannot be zero"))
+			}
+		}
+	}
+
+	// Consider removing this once moves out of preview
+	// Updating outboundType after cluster creation (PREVIEW)
+	// https://learn.microsoft.com/en-us/azure/aks/egress-outboundtype#updating-outboundtype-after-cluster-creation-preview
+	if err := webhookutils.ValidateImmutable(
+		field.NewPath("Spec", "Template", "Spec", "OutboundType"),
+		old.Spec.Template.Spec.OutboundType,
+		mcp.Spec.Template.Spec.OutboundType); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	if errs := mcp.validateVirtualNetworkTemplateUpdate(old); len(errs) > 0 {
+		allErrs = append(allErrs, errs...)
+	}
+
+	if errs := mcp.validateAPIServerAccessProfileTemplateUpdate(old); len(errs) > 0 {
+		allErrs = append(allErrs, errs...)
+	}
+
+	if len(allErrs) == 0 {
+		return nil, mcp.validateManagedControlPlaneTemplate(mcpw.Client)
+	}
+
+	return nil, apierrors.NewInvalid(GroupVersion.WithKind("AzureManagedControlPlaneTemplate").GroupKind(), mcp.Name, allErrs)
```

Comment:
now that we have consts for kinds, would be good to also use constants for these new kinds 

Can be a follow up since that other PR just merged